api/v1/client/client.go

```diff
@@ -72,15 +72,52 @@ func WithHTTPClient(httpClient *http.Client) Option {
 	})
 }
 
-// WithLogger sets the specifield logger to the client.
+// WithLogger sets the specified logger to the client.
 func WithLogger(logger *logrus.Entry) Option {
 	return Option(func(c *Client) {
 		c.logger = logger
 	})
 }
 
+// CallAPI executes the desired REST API request.
+// it's expected that the body and out are the structs that follows the JSON:API
+func (c *Client) CallAPI(ctx context.Context, method string, rel *url.URL, body, out interface{}) (err error) {
```

Comment:
yes, there are some usages in the other projects that imports k6

Reply:
Do we plan to unexport in https://github.com/grafana/k6/pull/2313?

Reply:
No, there is no plan to make this non-exported. If we do so, the components that rely on this logic should repeat the call method.

Instead of this, we can think about extracting the client code to the separated package, and there this can be internal, but I believe we should do that after the refactoring of the API.